src/bin/ion/commands/generate/templates/java/class.templ

```diff
@@ -65,60 +63,72 @@ public class {{ target_kind_name }} {
-                                                {% elif field.value_type is containing("ArrayList") %}
-                                                    {{ util_macros::read_as_sequence(field=field) }}
+                                                {% elif field_value is containing("ArrayList") %}
+                                                    {{ util_macros::read_as_sequence(field_value=field_value,field_name=field_name,type_store=type_store) }}
                                                 {% else %}
-                                                    {{ field.value_type }}.readFrom(reader);
+                                                    {{ field_value }}.readFrom(reader);
                                                 {% endif %}
                     break;
                 {% endfor %}
                 default:
-                    throw new IonException("Can not read field name:" + fieldName + " for {{ target_kind_name }} as it doesn't exist in the given schema type definition.");
+                    throw new IonException("Can not read field name:" + fieldName + " for {{ model.name }} as it doesn't exist in the given schema type definition.");
             }
         }
         reader.stepOut();
 
-        {{ target_kind_name }} {{ target_kind_name | camel }} = new {{ target_kind_name }}();
-        {% for field in fields -%}
-            {{ target_kind_name | camel }}.{{ field.name | camel }} = {{ field.name | camel }};
+        {{ model.name }} {{ model.name | camel }} = new {{ model.name }}();
+        {% for field_name, field_value in struct_info["fields"] -%}
+            {{ model.name | camel }}.{{ field_name | camel }} = {{ field_name | camel }};
         {% endfor %}
 
-        return  {{ target_kind_name | camel }};
+        return  {{ model.name | camel }};
     }
 
     /**
-     * Writes a {{ target_kind_name }} as Ion from an {@link IonWriter}.
+     * Writes a {{ model.name }} as Ion from an {@link IonWriter}.
      *
      * This method does not close the writer after writing is complete.
      * The caller is responsible for closing the stream associated with the writer.
      */
     public void writeTo(IonWriter writer) throws IOException {
          {# Writes `Structure` class with multiple fields based on `field.name` as an Ion struct #}
          writer.stepIn(IonType.STRUCT);
-         {% for field in fields %}
-             writer.setFieldName("{{ field.name }}");
-             {% if field.value_type | is_built_in_type == false %}
-                 {% if field.value_type is containing("ArrayList") %}
-                    {{ util_macros::write_as_sequence(field=field) }}
+         {% for field_name, field_val in struct_info["fields"] %}
+             {% set field_value = field_val.0 | to_string %}
+             writer.setFieldName("{{ field_name }}");
+             {% if field_value | is_built_in_type == false %}
+                 {% if field_value is containing("ArrayList") %}
```

Comment:
Is this a proxy test for something else? Would an `is_sequence` test do the same thing?